waku/common/tokenbucket.nim

```diff
@@ -0,0 +1,61 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import chronos
+
+## This is an extract from chronos/ratelimit.nim due to the found bug in the original implementation.
+## Unfortunately that bug cannot be solved without harm the original features of TokenBucket class.
+## So, this current shortcut is used to enable move ahead with nwaku rate limiter implementation.
+type TokenBucket* = ref object
+  budget*: int
+  budgetCap: int
+  lastTimeFull: Moment
+  fillDuration: Duration
+
+proc update(bucket: TokenBucket, currentTime: Moment) =
+  if bucket.fillDuration == default(Duration):
```

Comment:
Could you kindly add a comment elaborating on the purpose of this `proc` and when is being used?

Reply:
Sure!